### PR TITLE
change weight_decay for DecoupledWeightDecay optimizers in tensorflow

### PR DESCRIPTION
**Pull request by vbvg2008**

### CODE REVIEW

File: fastestimator/backend/load_model.py [hunk 1]
```diff
@@ -57,7 +58,10 @@ def load_model(model: Union[tf.keras.Model, torch.nn.Module], weights_path: str,
             with open(optimizer_path, 'rb') as f:
                 state_dict = pickle.load(f)
             model.current_optimizer.set_weights(state_dict['weights'])
-            set_lr(model, state_dict['lr'])
+            weight_decay = None
+            if isinstance(model.current_optimizer, tfa.optimizers.DecoupledWeightDecayExtension):
```
Comment (TortoiseHam) [inline fastestimator/backend/load_model.py:62]: check w/ mixed precision optimizer

File: fastestimator/backend/save_model.py [hunk 2]
```diff
@@ -64,7 +65,12 @@ def save_model(model: Union[tf.keras.Model, torch.nn.Module],
             assert model.current_optimizer, "optimizer does not exist"
             optimizer_path = os.path.join(save_dir, "{}_opt.pkl".format(model_name))
             with open(optimizer_path, 'wb') as f:
-                pickle.dump({'weights': model.current_optimizer.get_weights(), 'lr': get_lr(model)}, f)
+                saved_data = {'weights': model.current_optimizer.get_weights(), 'lr': get_lr(model)}
+                if isinstance(model.current_optimizer, tfa.optimizers.DecoupledWeightDecayExtension) or hasattr(
```
Comment (TortoiseHam) [inline fastestimator/backend/save_model.py:69]: if we're going to use a hasattr check should we just replace all of this with a hasattr(weight_decay) ?

File: fastestimator/backend/set_lr.py [hunk 3]
```diff
@@ -36,12 +39,21 @@ def set_lr(model: Union[tf.keras.Model, torch.nn.Module], lr: float):
     Args:
         model: A neural network instance to modify.
         lr: The learning rate to assign to the `model`.
+        weight_decay: The weight decay parameter, this is only relavent when using `tfa.DecoupledWeightDecayExtension`.
```
Comment (TortoiseHam) [inline fastestimator/backend/set_lr.py:42]: relevant

Comment (vbvg2008) [reply fastestimator/backend/set_lr.py]: fixed

**Review by TortoiseHam — APPROVED** [review]
LG

**Pull request merged**